apps/scan/frontend/src/screens/scan_error_screen.tsx

```diff
@@ -26,50 +27,117 @@ export function ScanErrorScreen({
   scannedBallotCount,
   restartRequired = false,
 }: Props): JSX.Element {
-  const errorMessage = (() => {
-    if (!error) return undefined;
-    assert(
-      error !== 'double_feed_calibration_timed_out' && // Only used in double feed calibration
-        error !== 'scanner_diagnostic_failed' && // Only used in ScannerDiagnosticScreen
-        error !== 'outfeed_blocked' // Only used in ScanJamScreen
-    );
+  assert(
+    error !== 'double_feed_calibration_timed_out' && // Only used in double feed calibration
+      error !== 'scanner_diagnostic_failed' && // Only used in ScannerDiagnosticScreen
+      error !== 'outfeed_blocked' // Only used in ScanJamScreen
+  );
+
+  const {
+    title,
+    errorMessage,
+    pollWorkerMessage,
+  }: {
+    title: JSX.Element;
+    errorMessage: JSX.Element;
+    pollWorkerMessage: 'needs-help' | 'might-need-help' | 'none';
+  } = (() => {
+    if (
+      restartRequired ||
+      !error ||
+      error === 'paper_status_timed_out' ||
+      error === 'scanning_timed_out' ||
+      error === 'unexpected_paper_status' ||
+      error === 'unexpected_event' ||
+      error === 'client_error'
+    ) {
```

Comment:
One of the most confusing parts of this screen was that there were a few different paths to the same or similar errors. `restartRequired` indicates a restart is required, which is associated with a serious of restart required error messages, but it was not clear to me whether `restartRequired` could be triggered with a different error or an undefined error. Without knowing, I erred on the side of the refactor not affecting previous behavior.